src/Microsoft.ML.KMeansClustering/KMeansPlusPlusTrainer.cs

```diff
@@ -45,34 +45,54 @@ public enum InitAlgorithm
         }
 
         [BestFriend]
-        internal static class Defaults{
+        internal static class Defaults
+        {
             /// <value>The number of clusters.</value>
-            public const int K = 5;
+            public const int ClustersCount = 5;
         }
 
-        public class Arguments : UnsupervisedLearnerInputBaseWithWeight
+        public class Options : UnsupervisedLearnerInputBaseWithWeight
```

Comment:
>Options [](start = 21, length = 7)

Just an observation.
I found it's funny what we diverge during ITransformer/IEstimator conversion and for what used to be transforms we come up with ColumnInfo and we no longer rely on arguments. (We still have conversion from arguments to columnInfo for EntryPoints)

Eventually half of our entrypoints would become options, and other half remain arguments.

Reply:
good point... and a bummer. Does it make sense to have a plan to reconcile?

---
In reply to: [248412000](https://github.com/dotnet/machinelearning/pull/2158#discussion_r248412000) [](ancestors = 248412000)